src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -97,28 +97,14 @@ String getServerVersion() throws SQLException {
     }
 
     @Override
-    public Statement createStatement() throws SQLException {
-        checkConnection();
-        try {
-            return new MongoStatement(this, currentDB, relaxed);
-        } catch (IllegalArgumentException e) {
-            throw new SQLException(e);
-        }
-    }
+    public abstract Statement createStatement() throws SQLException;
```

Comment:
I have a general observation on this function and other ones like it: createStatement() is part of the Connection interface and does not need to be redeclared here.  The subclass will still need to implement all the interface functions.
It may be cleaner to remove the Overriden functions that are redeclared as abstract.

Reply:
That's a very good point, I'll do an audit of which ones don't need to be redeclared and make the updates 👍

Reply:
Done

Reply:
Thanks for making the change, it's clearer now what functionality is provided in the abstract class.